zoo/auditing/utils.py

```diff
@@ -10,5 +18,154 @@ def create_git_issue(issue, user_name, reverse_url):
     issue.save()
 
 
-def get_md5_hash(string):
-    return hashlib.md5(string).hexdigest()
+def generate_random_key(length=512):
+    return "".join(
+        random.SystemRandom().choice(string.ascii_uppercase + string.digits)
```

Comment:
Any reason for not using something like `uuid`?

Reply:
I've read this post - https://stackoverflow.com/questions/2257441/random-string-generation-with-upper-case-letters-and-digits/23728630#23728630 and thought it would be better

Reply:
But I didn't pay that much attention to it, there is also apparently a `secrets` module since python3.6